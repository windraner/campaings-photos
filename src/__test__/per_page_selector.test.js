import React from 'react';
import PerPageSelector from '../components/main_container/bottom_bar/per_page_selector/per_page_selector';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

const mock = {
  value: 25,
  stateHandler() {},
};

describe('PerPageSelector', () => {
  it('matches the snapshot', () => {
    const tree = shallow(<PerPageSelector {...mock} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
