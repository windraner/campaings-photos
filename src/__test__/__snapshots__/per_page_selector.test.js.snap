// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PerPageSelector matches the snapshot 1`] = `
<div
  className="d-flex"
>
  <div
    className="campaigns-photos__select-text d-flex align-items-center mr-2"
  >
    Results per page:
  </div>
  <div
    className="position-relative"
  >
    <select
      className="campaigns-photos__select"
      onChange={[Function]}
      value={25}
    >
      <option
        className="campaigns-photos__select-option"
        key="0"
      >
        25
      </option>
      <option
        className="campaigns-photos__select-option"
        key="1"
      >
        50
      </option>
      <option
        className="campaigns-photos__select-option"
        key="2"
      >
        200
      </option>
    </select>
    <span
      className="campaigns-photos__select-icon"
    />
  </div>
</div>
`;
